core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1426,7 +1426,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * supported for Hadoop-supported filesystems.
    */
   def addFile(path: String, recursive: Boolean): Unit = {
-    val uri = new URI(path)
+    val uri = new Path(path).toUri
```

Comment:
I just check the documentation and tests. It seems Windows paths are being tested in https://github.com/apache/hadoop/blob/branch-2.7.2/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestPath.java as `toString` it based on `URI` instance in `Path` which `toUri` will return directly, I guess it'd be safe.

I could not find the explicit mention about Windows path in the documenation https://hadoop.apache.org/docs/stable2/api/index.html